manifest/v1alpha/budgetadjustment/validation_test.go

```
@@ -375,3 +431,104 @@ func validBudgetAdjustment() BudgetAdjustment {
+		{
+			name:          "single occurrence rrule returns no error",
+			rule:          "FREQ=MINUTELY;COUNT=1",
+			expectedError: "",
+		},
+		{
+			name:          "two times minutely occurrence rrule returns error",
+			rule:          "FREQ=MINUTELY;COUNT=2",
+			expectedError: "interval must be at least 60 minutes for minutely frequency",
+		},
+		{
+			name:          "hourly frequency returns no error",
+			rule:          "FREQ=HOURLY;INTERVAL=1",
+			expectedError: "",
+		},
+		{
+			name:          "daily frequency returns no error",
+			rule:          "FREQ=DAILY;INTERVAL=1",
+			expectedError: "",
+		},
+		{
+			name:          "frequency greater than hourly in minutes no error",
+			rule:          "FREQ=MINUTELY;INTERVAL=61;COUNT=10",
+			expectedError: "",
+		},
+		{
+			name:          "frequency greater than hourly in seconds no error",
+			rule:          "FREQ=SECONDLY;INTERVAL=3600;COUNT=10",
+			expectedError: "",
+		},
+		{
+			name:          "frequency greater than hourly in seconds no error",
+			rule:          "FREQ=SECONDLY;INTERVAL=3600",
+			expectedError: "",
+		},
+		{
+			name:          "frequency shorter than hourly in seconds returns error",
+			rule:          "FREQ=SECONDLY;INTERVAL=3500;COUNT=10",
+			expectedError: "interval must be at least 3600 seconds for secondly frequency",
+		},
+		{
+			name:          "minutely with by hour returns error",
+			rule:          "FREQ=MINUTELY;BYHOUR=6,8,9,10,11,12,13,14,15,16;COUNT=10",
+			expectedError: "interval must be at least 60 minutes for minutely frequency",
+		},
+		{
+			name:          "minutely with by hour returns error",
+			rule:          "FREQ=MINUTELY;INTERVAL=10;BYHOUR=6,8,9,10,11,12,13,14,15,16",
+			expectedError: "interval must be at least 60 minutes for minutely frequency",
+		},
+		{
+			name:          "hourly with by minute returns no error",
+			rule:          "FREQ=HOURLY;BYHOUR=6,8,9,10,11,12,13,14,15,16;BYMINUTE=6,8,9,10,11,12,13,14,15,16,59;COUNT=10",
+			expectedError: "",
```

this should throw an error